Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabSystemComponent.cpp

```diff
@@ -753,6 +754,80 @@ namespace AzToolsFramework
             }
         }
 
+        bool PrefabSystemComponent::AreDirtyTemplatesPresent(TemplateId templateId)
+        {
+            auto prefabTemplate = FindTemplate(templateId);
+
+            if (!prefabTemplate.has_value())
+            {
+                AZ_Assert(false, "Template with id %llu is not found", templateId);
+                return false;
+            }
+
+            if (IsTemplateDirty(templateId))
+            {
+                return true;
+            }
+
+            auto linkIds = prefabTemplate->get().GetLinks();
+
+            for (auto linkId : linkIds)
+            {
+                auto linkIterator = m_linkIdMap.find(linkId);
+                if (linkIterator != m_linkIdMap.end())
+                {
+                    return AreDirtyTemplatesPresent(linkIterator->second.GetSourceTemplateId());
+                }
+            }
+            return false;
+        }
+
+        void PrefabSystemComponent::SaveAllDirtyTemplates(TemplateId templateId)
+        {
+            AZStd::set<AZ::IO::PathView> dirtyTemplatePaths;
+            GetDirtyTemplatePaths(templateId, dirtyTemplatePaths);
+
+            for (auto dirtyTemplatePath : dirtyTemplatePaths)
+            {
+                auto dirtyTemplateIterator = m_templateFilePathToIdMap.find(dirtyTemplatePath);
+                if (dirtyTemplateIterator == m_templateFilePathToIdMap.end())
+                {
+                    AZ_Assert(false, "Template id for template with path '%s' is not found.", dirtyTemplatePath);
+                }
+                else
+                {
+                    m_prefabLoader.SaveTemplate(dirtyTemplateIterator->second);
+                }
+            }
+        }
+
+        void PrefabSystemComponent::GetDirtyTemplatePaths(TemplateId templateId, AZStd::set<AZ::IO::PathView>& dirtyTemplatePaths)
```

Comment:
Prefer return via value here and unordered_set as it has faster lookup, also could be named like:
AZStd::unordered_set<AZ::IO::PathView> PrefabSystemComponent::GetDirtyChildTemplatePaths(TemplateId rootTemplateId)

If it's expected that you would want to call this function multiple times reusing the dirtyTemplatePaths, in which case output parameter is fine

Reply:
sounds good